python_bindings/correctness/compile_to.py

```diff
@@ -37,7 +37,10 @@ def main():
 
         f.compile_to_file(os.path.join(tmpdir, "f_all"), args)
         assert os.path.isfile(os.path.join(tmpdir, "f_all.h"))
-        assert os.path.isfile(os.path.join(tmpdir, "f_all.o"))
+        if hl.get_target_from_environment().os == hl.TargetOS.Windows:
+            assert os.path.isfile(os.path.join(tmpdir, "f_all.obj"))
+        else:
+            assert os.path.isfile(os.path.join(tmpdir, "f_all.o"))
```

Comment:
This would be good (also along with the PythonExtensions.cpp changes) to pull into its own PR.

Reply:
Done.  Created https://github.com/halide/Halide/pull/4743